include/swift/SIL/SILOpenedArchetypesTracker.h

```diff
@@ -44,12 +36,6 @@ class SILOpenedArchetypesTracker : public DeleteNotificationHandler {
   SILOpenedArchetypesTracker(SILOpenedArchetypesTracker &Tracker)
       : SILOpenedArchetypesTracker(Tracker.F, Tracker) {}
```

Comment:
Should this be a `T&&` constructor instead of a `T&` constructor? Overloading on const/non-const lvalue ref feels brittle to me. I think @eeckstein or @swiftix wrote this code originally.